azurelinuxagent/common/utils/restutil.py

```diff
@@ -78,6 +77,7 @@
 
 HTTP_PROXY_ENV = "http_proxy"
 HTTPS_PROXY_ENV = "https_proxy"
+HTTP_USER_AGENT = "{0}/{1}".format(AGENT_NAME, CURRENT_VERSION)
```

Comment:
I think you actually want the `GOAL_STATE_AGENT_VERSION`. There is some complexity because you are not guaranteed it is correct, however you might just include both for simplicity.

Reply:
Agreed.

I am more concerned with having WALinuxAgent than I am with the version at the moment.  Let's see if this change actually brings any benefit for PIR, and we can iterate if necessary.